src/Parser/Hunter/Marksmanship/Modules/Traits/Bullseye.js

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import SPELLS from 'common/SPELLS';
+import StatisticBox from "Main/StatisticBox";
+import SpellIcon from "common/SpellIcon";
+import { formatNumber, formatPercentage } from "common/format";
+import CooldownTracker from '../Features/CooldownTracker';
+
+class Bullseye extends Analyzer{
+  executeTimestamp;
+  bullseyeResets = 0; //only resets when boss < 20% health, so resets we can confirm shouldn't have happened
+  bullseyeInstances = [];
+  saveForExecuteThreshhold;
+  bossIDs = [];
+
+  static dependencies = {
+    cooldownTracker : CooldownTracker,
+  }
+  on_initialized(){
+    this.owner.report.enemies.forEach(enemy => {
+      if (enemy.fights[0].id === this.owner.fight.id && enemy.type === "Boss"){ //ensure that we only count Bosses for checking if bullseye was reset during execute
```

Comment:
I think this will only pass if the enemy was first see in the selected fight, but it will probably break on subsequent pulls.

Reply:
Sorry, not 100% what you mean by this. Are there any cases where the enemy isn't first seen in the selected fight? Repeat bosses aren't really a thing, are they? Or do you mean if there are heroic and mythic kills in the same log?